stripe/src/main/java/com/stripe/android/cards/RemoteCardAccountRangeSource.kt

```diff
@@ -25,16 +33,35 @@ internal class RemoteCardAccountRangeSource(
         return cardNumber.bin?.let { bin ->
             mutableLoading.value = true
 
-            val accountRanges =
-                stripeRepository.getCardMetadata(bin, requestOptions)?.accountRanges.orEmpty()
+            val accountRanges = stripeRepository.getCardMetadata(bin, requestOptions)?.accountRanges.orEmpty()
             cardAccountRangeStore.save(bin, accountRanges)
 
             mutableLoading.value = false
 
-            accountRanges
-                .firstOrNull { (binRange) ->
-                    binRange.matches(cardNumber)
+            when {
+                accountRanges.isNotEmpty() -> {
+                    val matchedAccountRange = accountRanges
+                        .firstOrNull { (binRange) ->
+                            binRange.matches(cardNumber)
+                        }
+
+                    if (matchedAccountRange == null) {
+                        fireMissingRangeRequest()
```

Comment:
will this fire if it's actually a bad PAN that doesn't match any? I configured on iOS so it has to at least match one of the hard-coded bin ranges and pass a luhn check before I send this event

Reply:
Makes sense. I think just doing the Luhn check seems sufficient. I'll add that.